jlc_lib/generate_bom.py

```diff
@@ -42,25 +42,59 @@ def GenerateBOM(input_filename, output_filename):
   num_groups_found = 0
   for group in grouped:
     refs = []
+    lcsc_part_numbers = []
+    footprints = []
+
     for component in group:
-        refs.append(component.getRef())
-        c = component
-
-    # Get the field name for the LCSC part number.
-    lcsc_part_number = ""
-    for field_name in c.getFieldNames():
-      field_value = c.getField(field_name)
-      if LCSC_PART_NUMBER_MATCHER.match(field_value):
-        lcsc_part_number = field_value
-
-    # Skip groups without LCSC part number.
-    if lcsc_part_number == "":
+      refs.append(component.getRef())
+      c = component
+
+      # Get the field name for the LCSC part number.
+      for field_name in c.getFieldNames():
+        field_value = c.getField(field_name)
+
+        if LCSC_PART_NUMBER_MATCHER.match(field_value):
+          lcsc_part_numbers.append(field_value)
+        else:
+          lcsc_part_numbers.append(None)
+
+      footprints.append(c.getFootprint())
+
+
+    # Check part numbers for uniqueness
+    lcsc_part_numbers_set = set(lcsc_part_numbers)
+    lcsc_part_numbers_set_without_none = lcsc_part_numbers_set - set([None])
+    if len(lcsc_part_numbers_set_without_none) == 0:
+      if opts.warn_no_partnumber:
+        logging.warning("No LCSC part number found for components {}".format(",".join(refs)))
       continue
+    elif len(lcsc_part_numbers_set_without_none) != 1:
+      logging.error("Components {components} from same group have different LCSC part numbers: {partnumbers}".format(
+          components = ", ".join(refs),
+          partnumbers = ", ".join(lcsc_part_numbers_set_without_none)))
+      return False
+    lcsc_part_number = list(lcsc_part_numbers_set)[0]
+
+    if (not opts.assume_same_lcsc_partnumber) and (None in lcsc_part_numbers_set):
+      logging.error("Components {components} from same group do not all have LCSC part number {partnumber} set. Use --assume-same-lcsc-partnumber to ignore.".format(
+          components = ", ".join(refs),
+          partnumber = lcsc_part_number))
+      return False
 
+    # Check footprints for uniqueness
+    footprints_set = set(footprints)
```

Comment:
Same here. Just build the set incrementally.

Reply:
Fixed, see above.